RetailCoder.VBE/UI/Command/ExportAllCommand.cs

```diff
@@ -0,0 +1,77 @@
+using System.IO;
+using System.Windows.Forms;
+using NLog;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+using Rubberduck.Navigation.CodeExplorer;
+using Rubberduck.UI.CodeExplorer.Commands;
+
+namespace Rubberduck.UI.Command
+{
+    [CodeExplorerCommand]
+    public class ExportAllCommand : CommandBase 
+    {
+        private readonly IVBE _vbe;
+        private IFolderBrowserFactory _factory;
+
+        public ExportAllCommand(IVBE vbe, IFolderBrowserFactory folderBrowserFactory) : base(LogManager.GetCurrentClassLogger())
+        {
+            _vbe = vbe;
+            _factory = folderBrowserFactory;
+        }
+
+        protected override bool EvaluateCanExecute(object parameter)
+        {
+            if (parameter == null)
+            {
+                return Evaluate(_vbe.ActiveVBProject);
+            }
+            else if (parameter is CodeExplorerProjectViewModel)
+            {
+                var node = (CodeExplorerProjectViewModel)parameter;
+                return Evaluate((IVBProject)node.Declaration.Project);
```

Comment:
They're pointers to the same object. The simplified syntax leverages null-conditional `.?`, and null-coalescing `??` operators. It's more efficient because there's fewer casting - other than that it's equivalent.